File: q1312/death10_1.cs
using System;
class Z{static void Main(){
string a="GTGT",b=a+a+"GTT"+a+"TTGTT",c="TTTFGTTTT"+b+"TWWTTWG"+b+"FGGWMTMTGGTWGGTTGGTTMWGGWGTTMGGM";foreach(var e in c)Console.Write(e=='T'?"tree":e=='F'?"fire":e=='G'?"gold":e=='W'?"water":"moon");
}}

File: q1312/death10_1.py
a='GT'
b=a*5+'T'+a*2+'TTGTT'
print(''.join({'T':'tree','F':'fire','G':'gold','W':'water','M':'moon'}[e] for e in 'TTTF'+a+'TTT'+b+'TWWTTWG'+b+'FGGWMTMTG'+a+'WG'+a+'TG'+a+'TMWGGW'+a+'TMGGM'))